features/importers/myhelsinki_places/tests/test_import_tags.py

```diff
@@ -48,3 +55,30 @@ def test_tags_are_linked_to_features(requests_mock, importer, places_response):
 
     assert feature.tags.count() == 1
     assert tag in feature.tags.all()
+    assert feature.feature_tags.first().source == FeatureTagSource.MAPPING
+
+
+@pytest.mark.parametrize(
+    "tag_id,tag_name,expected",
+    [
+        ("manualtag", "manual", 2),  # Importer doesn't assign this
+        ("ahti:tag:island", "saaristo", 1),  # Importer would assign this
+    ],
+)
+def test_manually_set_tags_are_not_touched_when_importing(
+    tag_id, tag_name, expected, requests_mock, importer, places_response
+):
+    """Manually tag to a feature and asses that it's not touched by the importer.
+
+    Only FeatureTag objects with source as MAPPING are automatically managed.
+    FeatureTag.source should not be changed from MANUAL to MAPPING.
+    """
+    requests_mock.get(PLACES_URL, json=places_response)
+    tag = TagFactory(id=tag_id, name=tag_name)
+    feature = FeatureFactory(source_type=importer.get_source_type(), source_id="2792")
+    feature.tags.add(tag, through_defaults={"source": FeatureTagSource.MANUAL})
+
+    importer.import_features()
+
+    assert feature.tags.count() == expected
+    assert feature.feature_tags.get(tag_id=tag_id).source == FeatureTagSource.MANUAL
```

Comment:
As discussed yesterday, I have some problems with how this test reads. One problem is the parameter comments "Importer doesn't/would assign this", it sounds like the given tags are processed somehow differently. One of them being called "manualtag" implies the other one isn't a manual one. In fact, both become manual tags, and the expected value only changes because "saaristo" is already defined elsewhere in tag configurations.

Would it be possible to write this test to be more explicit in it's behavior? Like "given one imported tag and one different manual tag, the expectation after importing is ..."

Is it necessary to have the expected number as a parameter? Would it be okay to count the number of distinct id's to begin with (from tag config + parameters) and after importing (from resulting tags) and make sure no manually defined tags are changed? Just a thought.